sqlmodel/_compat.py

```diff
@@ -80,6 +82,10 @@ def partial_init() -> Generator[None, None, None]:
     from pydantic_core import PydanticUndefined as Undefined
     from pydantic_core import PydanticUndefinedType as UndefinedType
 
+    PYDANCTIC_FIELD_GET_DEFAULT_REQUIRES_VALIDATED_DATA = (
+        "validated_data" in inspect.signature(FieldInfo.get_default).parameters
+    )
```

Comment:
Can't a version check be done instead? I think it would clearer with a comment:

```suggestion
    # Pydantic 2.10 added support for default factories taking validated data:
    PYDANCTIC_FIELD_GET_DEFAULT_REQUIRES_VALIDATED_DATA = IS_PYDANTIC_210
    )
```